warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -1350,9 +1355,32 @@ protected void loadQueryModel(MetadataHelper helper, ShardQueryConfiguration con
     }
 
     protected Scheduler getScheduler(ShardQueryConfiguration config, ScannerFactory scannerFactory) {
+        // ony optimize if unique fields are not requested
+        if (isUseDocumentScheduler() && config.getUniqueFields().isEmpty() && !isCheckpointable() && !isFullTableScanEnabled()) {
```

Comment:
can you add a blurb here highlighting why this can't work with unique fields?

Reply:
This is kind of interesting. I originally suspected that the unique transform on the query iterator would still be necessary to avoid overwhelming the unique transform in the webservice, but I ran a test where the document scheduler would handle a query with a unique function and saw no discernable difference. 

I would like to do a more focused test with unique functions in the near future. Ideally the SimpleQueryVisitor doesn't have to deal with external state like the ShardQueryConfig. Given that, I think it's appropriate to check for unique fields here. Although I hope that check goes away entirely soon.